astropy/io/fits/tests/test_header.py

```diff
@@ -2371,6 +2371,16 @@ def test_header_fromstring_bytes(self):
         assert pri_hdr == pri_hdr_from_bytes
         assert pri_hdr.tostring() == pri_hdr_from_bytes.tostring()
 
+    def test_set_keyword_with_space(self):
+        """
+        Regression test for https://github.com/astropy/astropy/issues/10479
+        """
+        hdr = fits.Header()
+        hdr['KEY2 '] = 2
+        hdr['KEY2 '] = 4
+        assert len(hdr) == 1
+        assert hdr['KEY2 '] == 4
```

Comment:
Just maybe to make things weirder, replace:
```suggestion
        assert hdr['KEY2  '] == 4
```
to have a different number of trailing spaces.

Reply:
In that case I'd recommend to keep both, just in case some even weirder bug sneaks in that leaves it at
```
hdr['KEY2 '] = 2
hdr['KEY2  '] = 4
```

Reply:
Good ideas, thanks, I updated the test.

Reply:
@dhomeier That would be detected by `assert len(hdr) == 1`